File: C#/Easy/BestTimeToBuyAndSellStock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int left = 0;
        int max = 0;

        for (int right = 0; right < prices.Length; right++)
        {
            int diff = prices[right] - prices[left];
            if (diff < 0)
                left = right;
            max = Math.Max(max, diff);
        }

        return max;
    }
}

/*

  Kadanes algorithm (two pointers)
  
  Time Complexity: O(n)
  Space Complexity: O(1)
  
  Where n is the length of prices

*/

File: Swift/Easy/BestTimeToBuyAndSellStock.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var result = 0
        var left = 0
        for right in 1..<prices.count {
            let profit = prices[right] - prices[left]
            if (profit < 0) {
                left = right
            }
            result = max(result, profit)
        }

        return result
    }
}

/*

    Time: O(n)
    Space: O(1)

*/
